### DISCUSSION TITLE
提几个建议

### DISCUSSION THREAD

**Discussion by oTnTh** [morlay/genshintools#5] [General]
NGA看到的，不过那边也不像个讨论事情的地方，就写这里了。

1、考虑过发布PC版么？

我尝试了一下网上流传的手机端断网的法子，在我的iPhone上无法获得祈愿地址，表现是直接开始转圈圈，然后回到游戏登录界面。

如果安卓也是如此的话，目前能更新祈愿纪录的就只有桌面端了。（桌面导入 => 同步 => 移动端）

2、考虑发布iOS版么？

我看到你说的开发者账号的问题了，不能上架，直接放出ipa其实也可以，毕竟cookie和webdav都很硬核了，通过自签名安装也是可以的。

3、还是祈愿纪录导入的问题。

如果没有PC版的话，那解决从其他app导入的问题也可以。

这里有个统一可交换祈愿记录标准：https://github.com/DGP-Studio/Snap.Genshin/wiki/StandardFormat

我自己把别的app导出的这个格式的json，转换成genshintools，算md5sum，存倒webdav上覆盖，这个过程太复杂了，肯定不是一般用户能搞定的。

flutter我完全不了解，只是去官网看了看说安卓ios和桌面同时支持，如果很麻烦的话不做也没关系。（

NGA上总删帖，老回答一模一样的问题感觉你也满辛苦的，如果要建群什么的会不会好一点。

最后，非常感谢。

**Comment by oTnTh** [comment]
还有一件事忘记写了。

gitee下载太麻烦了，手机端更新的时候还要登录就更麻烦。我看到某些app的更新包是放在jsdelivery上的，目前为止看来没出啥问题，你要不要也试试。

**Reply by morlay** [reply]
抽卡记录格式和 gitee 的问题已经改了。

没用 jsdelivery，缓存坑，github 发布，走国内加速代理。

**Comment by morlay** [comment]
> 1、考虑过发布PC版么？

Flutter for windows 刚发布，有什么坑不清楚，另外，mobile 和 pc 的交互逻辑是不一样，精力有限不考虑去折腾了这个支持了。

如果只是抽卡记录的问题
按照其他工具的说法 https://feixiaoqiu.com/rank_url_upload_init/
PC 上的 log 链接，拿过来也是可以直接用的。

> 2、考虑发布iOS版么？

不发布，确实就是开发者账号的问题，再加上版权一定会出事。
源码支持自行编译，因为我本就是在 ios 模拟题上调试开发的。

> 这里有个统一可交换祈愿记录标准：https://github.com/DGP-Studio/Snap.Genshin/wiki/StandardFormat

这个格式的迁移在计划内，因为要支持多账户，都会加一层 wrap ` { "<uid>":  specObj }`

另外，sum 那个文件直接删了就成，因为只是为了避免重复同步而过渡消耗坚果云的流量。
只需要替换数据。

> gitee下载太麻烦了，手机端更新的时候还要登录就更麻烦。我看到某些app的更新包是放在jsdelivery上的，目前为止看来没出啥问题，你要不要也试试。

jsdelivery 其实也会被 gfw 的，时不时的。不过也好，我找时间换上去吧。
这样还省的我手动发布，搞到 github actions 自动点

**Comment by oTnTh** [comment]
关于抽卡纪录，我觉得这东西的社交属性远大于工具属性，无非是朋友聊天的时候说起你欧了我非了的时候有用。鉴于手机端现在好像没有法子（是没有吧？）直接更新抽卡纪录，所以就只能从pc上拿数据了。 

那么对用户来说相对友好一点的法子大概有两个：

1、genshintools做一个桌面版，在桌面上更新，webdav同步，再同步到手机上；
2、直接从其他流行的桌面版工具，比如牛杂、sg里抓数据，webdav同步或者直接就wifi传输导入手机。

方法2需要写一个不相干的桌面端程序，所以我才想如果如果flutter的桌面支持靠谱的话，不如直接发布一个桌面版。

（其实是觉得抽卡纪录的部分你写都写了，就这么扔着未免可惜，不过就社交属性来说，现在的genshintools的信息展现方式就太老实了点。比如没有出货概率，最新抽出来的五星也没排最前面等等。）

关于ios版，我的意思是说，如果build的时候不麻烦，顺便生成一个ipa文件就好。愿意折腾的人可以自己下载回去签名安装，虽然用起来也挺麻烦的，但也算多了点被人看到的机会吧。

**Reply by morlay** [reply]
社交属性，交给其他工具了。
我只想做个 抽卡规划辅助，作为月卡玩家。
反正现在兼容 uigf。

而且原神我当单机养老游戏玩的。
所以不会给太多精力做额外功能了。

另外，ios 不是说打包一个 ipa 就能用的，装到其他设备还得用开发者账号签名。

**Reply by morlay** [reply]
另 Android 一直能拿到抽卡地址

**Comment by oTnTh** [comment]
再就是，要不要考虑改个名字啊。莫娜占卜铺、椰羊、刻记牛杂店，工具箱这个名字也过于实在了，而且也不利于传播的说。

“我发现个很好用的手机app。”

“真的么，叫啥？”

“原神工具箱…”

**Reply by morlay** [reply]
传播小反而最好。
毕竟用到了解包数据。
我已经被举报没了一个玩具了。

**Comment by oTnTh** [comment]
如果只是安装ipa的话，倒是没那么难。不需要xcode，也不需要越狱啥的，虽然用起来确实有各种各样的麻烦和限制：

https://www.i4.cn/
https://sideloadly.io/

另外有些ocr扫描导出圣遗物信息的工具，比如这个：

https://github.com/wormtql/yas

不过目前导出的信息并不包括哪个角色装备了哪些圣遗物。如果你有兴趣的话，要不要去py一下，如果那边作者愿意改，就不需要手工输入直接导入信息就可以算伤害了。

**Reply by morlay** [reply]
那不就是麻烦么，还不能通知更新，所以还是不加了。
有需求，clone 代码，setup 下 flutter 的环境，跑一下就有了。

圣遗物格式就是 GOOD https://frzyc.github.io/genshin-optimizer/#/doc
之前 nga 其他玩家提的需求 https://github.com/daydreaming666/Amenoma
很早就改了。
转成 GOOD 就可以用的。

**Comment by oTnTh** [comment]
天目扫描起来太慢了，所以我一直都用的是yas。

我去yas提个issue，看看作者搭理不。

（感觉整理一下可以整出一篇FAQ了。

**Reply by morlay** [reply]
你要不做个占卜格式 转 good 呗

**Comment by oTnTh** [comment]
你说的是莫娜占卜铺导出的数据，再导入genshintools么？

我去试了下新版的[Amenoma](https://github.com/daydreaming666/Amenoma)，速度确实比以前更快了。虽然我选择不扫描四星及以下，他还是很执着地扫完了整个背包。

然后把天目导出的数据，整合到BlocGameData.json里，确实能用。不过BlocGameData.json.sum还是需要的，不然手机端的genshintools会不认。再就是我导入了800个多圣遗物，genshintools冷启动加载的时候能卡好一会儿。

**Reply by morlay** [reply]
sum 那个文件删掉就好了，不用自己再算的。

加载具体是哪一步慢？应用启动？还是圣遗物页面？

另，你的 BlocGameDate.json 多大了？

**Comment by oTnTh** [comment]
昨天第一次测试的时候，我导入了800多个圣遗物的信息，BlocGameData.json有400多k吧。打开app的时候点最下面一列的角色发现动不了，就想着应该是在加载数据，卡了好一会儿。

当时我应该是把webdav上的sum文件删掉了的，然后在手机里选了从webdav同步到本地，然后发现数据不对，webdav里的又变成手机端的数据了。不过也有可能是我点错了，下次再导入的时候我测试一下。

今天重新扫描了一下圣遗物，然后把满级的五星四星筛选出来。现在BlocGameData.json大概98k，161个圣遗物，这下倒是没卡了。

**Reply by morlay** [reply]
查词条等级没做缓存，数量多了复杂度就上去了。

空了再优化吧。最近有点忙了。

伤害计算数据还有一堆带条件的没处理，也还没想好要怎么弄，都快 200 多种了。

https://github.com/morlay/genshintools/blob/main/scripts/genshindb/openconfig.ts

**Comment by oTnTh** [comment]
现在这样就挺好了，genshintools让我知道原来刷了很久的试做没比精1阿莫斯差很多，就很满意了。

倒是怎么降低使用难度是个值得考虑的问题，给自己亲友安利了一下，解释老半天，就有点无奈。（
